app/internal/page/sign_your_lpa.go

```diff
@@ -0,0 +1,96 @@
+package page
+
+import (
+	"net/http"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+)
+
+type signYourLpaData struct {
+	App    AppData
+	Errors map[string]string
+	Lpa    *Lpa
+	Form   *signYourLpaForm
+}
+
+const (
+	CertificateProviderHasWitnessed = "cp-witnessed"
+	WantToApplyForLpa               = "want-to-apply"
+)
+
+type signYourLpaForm struct {
+	DonorSignatures []string
+}
+
+func SignYourLpa(tmpl template.Template, lpaStore LpaStore) Handler {
+	return func(appData AppData, w http.ResponseWriter, r *http.Request) error {
+		lpa, err := lpaStore.Get(r.Context(), appData.SessionID)
+		if err != nil {
+			return err
+		}
+
+		data := &signYourLpaData{
+			App: appData,
+			Lpa: lpa,
+			Form: &signYourLpaForm{
+				DonorSignatures: lpa.DonorSignatures,
+			},
+		}
+
+		if r.Method == http.MethodPost {
+			r.ParseForm()
+
+			data.Form = readSignYourLpaForm(r)
+			data.Errors = data.Form.Validate()
+
+			if len(data.Errors) == 0 {
+				lpa.DonorSignatures = data.Form.DonorSignatures
+				lpa.Tasks.ConfirmYourIdentityAndSign = TaskCompleted
+
+				if err = lpaStore.Put(r.Context(), appData.SessionID, lpa); err != nil {
+					return err
+				}
+
+				return appData.Lang.Redirect(w, r, appData.Paths.WitnessingYourSignature, http.StatusFound)
+			}
+		}
+
+		return tmpl(w, data)
+	}
+}
+
+func readSignYourLpaForm(r *http.Request) *signYourLpaForm {
+	r.ParseForm()
+
+	form := &signYourLpaForm{
+		DonorSignatures: []string{},
```

Comment:
why make this a list?

Reply:
It was so we could use the `checkbox` template which expects a list of items in order to function correctly (specifically the checked params). Like I said in my PR description, it does feel a bit redundant and I guess I could just use markup rather than relying on the template.